client/extensions/woocommerce/state/data-layer/action-list/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { getActionList } from 'woocommerce/state/action-list/selectors';
+import { actionListStepStart } from 'woocommerce/state/action-list/actions';
+import {
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export function handleStepStart( { dispatch, getState }, action ) {
+	const { stepIndex } = action;
+
+	const actionList = getActionList( getState() );
+	const step = actionList[ stepIndex ];
```

Comment:
That's a fair point. I was on the fence anyway. I'll make some changes for this.